client/state/sites/plans/selectors.js

```diff
@@ -170,7 +171,7 @@ export function isRequestingSitePlans( state, siteId ) {
 
 export function isCurrentPlanExpiring( state, siteId ) {
 	const currentPlan = getCurrentPlan( state, siteId );
-	const expiration = get( currentPlan, 'expiryMoment', null );
+	const expiration = currentPlan?.expiryDate && moment( currentPlan.expiryDate ).startOf( 'day' );
 	return expiration < moment().add( 30, 'days' );
```

Comment:
This logic is difficult to understand and possibly buggy. If there is no current plan or if it doesn't have `expiryDate` set, what is `isCurrentPlanExpiring` supposed to return? Currently it seems to be `true`, because `null < 1234` is `true`. Is that really desired?

We should be explicit about that:
```js
if ( ! currentPlan || ! currentPlan.expiryDate ) {
  return true; // or false 😊
}
```

Reply:
It's not clear to me either what it *should* be, so it's probably best to maintain the current behaviour. But I completely agree that it should be more explicit, so I went with your suggestion.